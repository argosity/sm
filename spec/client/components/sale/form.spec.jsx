import Sale from 'sm/models/sale';
import Show from 'sm/models/show';
import Form from 'sm/components/sale/form';

jest.mock('sm/extension', () => ({
    paymentsVendor: 'Square',
}));

describe('Payment Form', () => {
    let props;

    beforeEach(() => {
        props = {
            sale: new Sale({
                noCharge: true,
                show: new Show({
                    times: [{
                        occurs_at: new Date('2017-11-01T02:38:38.600Z'),
                        price: 12.3,
                    }],
                }),
            }),
            onComplete: jest.fn(),
        };
    });

    it('matches snapshot', () => {
        expect(<Form {...props} />).toMatchSnapshot();
    });

    it('sets sale values', () => {
        const form = mount(<Form {...props} />);
        form.find('input[name="qty"]').simulate(
            'change', { target: { value: 3 } },
        );
        form.find('input[name="name"]').simulate(
            'change', { target: { value: 'My Name' } },
        );
        form.find('input[name="email"]').simulate(
            'change', { target: { value: 'test@test.com' } },
        );
        form.find('input[name="phone"]').simulate(
            'change', { target: { value: '123-456-7890' } },
        );

        return form.instance().saveState().then((sale) => {
            expect(sale.qty).toEqual(3);
            expect(sale.name).toEqual('My Name');
            expect(sale.email).toEqual('test@test.com');
            expect(sale.phone).toEqual('123-456-7890');
        });
    });
});
