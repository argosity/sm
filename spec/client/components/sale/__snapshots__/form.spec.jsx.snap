// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Payment Form matches snapshot 1`] = `
<SaleForm
  heading=""
  onComplete={[Function]}
  sale={
    Sale {
      "created_at": undefined,
      "email": undefined,
      "errors": Object {},
      "id": undefined,
      "identifier": undefined,
      "isModel": true,
      "lastServerMessage": "",
      "message_id": undefined,
      "name": undefined,
      "noCharge": true,
      "payments": Array [],
      "phone": undefined,
      "qty": 1,
      "redemptions": Object {},
      "show": Show {
        "can_purchase": false,
        "capacity": undefined,
        "description": "",
        "errors": Object {},
        "external_url": undefined,
        "id": undefined,
        "identifier": "",
        "image": undefined,
        "isModel": true,
        "lastServerMessage": "",
        "message_id": undefined,
        "online_sales_halt_mins_before": undefined,
        "page": Object {},
        "page_images": Array [],
        "presenter": undefined,
        "presenter_id": undefined,
        "price": undefined,
        "sub_title": "",
        "syncInProgress": false,
        "ticket_instructions": undefined,
        "times": Array [
          ShowTime {
            "$nonSerializable": Object {
              "show": true,
            },
            "capacity": undefined,
            "errors": Object {},
            "id": undefined,
            "identifier": undefined,
            "isModel": true,
            "lastServerMessage": "",
            "occurs_at": 2017-11-01T02:38:38.600Z,
            "price": 12.3,
            "show": [Circular],
            "show_association_name": "times",
            "syncInProgress": false,
          },
        ],
        "title": "",
        "venue": undefined,
        "venue_id": undefined,
        "visible_during": "[2017-05-01T07:00:00.000Z,2017-05-09T06:59:59.999Z)",
      },
      "show_id": undefined,
      "show_identifier": undefined,
      "show_time_id": undefined,
      "syncInProgress": false,
      "tickets_url": undefined,
      "time": ShowTime {
        "$nonSerializable": Object {
          "show": true,
        },
        "capacity": undefined,
        "errors": Object {},
        "id": undefined,
        "identifier": undefined,
        "isModel": true,
        "lastServerMessage": "",
        "occurs_at": 2017-11-01T02:38:38.600Z,
        "price": 12.3,
        "show": Show {
          "can_purchase": false,
          "capacity": undefined,
          "description": "",
          "errors": Object {},
          "external_url": undefined,
          "id": undefined,
          "identifier": "",
          "image": undefined,
          "isModel": true,
          "lastServerMessage": "",
          "message_id": undefined,
          "online_sales_halt_mins_before": undefined,
          "page": Object {},
          "page_images": Array [],
          "presenter": undefined,
          "presenter_id": undefined,
          "price": undefined,
          "sub_title": "",
          "syncInProgress": false,
          "ticket_instructions": undefined,
          "times": Array [
            [Circular],
          ],
          "title": "",
          "venue": undefined,
          "venue_id": undefined,
          "visible_during": "[2017-05-01T07:00:00.000Z,2017-05-09T06:59:59.999Z)",
        },
        "show_association_name": "times",
        "syncInProgress": false,
      },
      "time_identifier": undefined,
      "token": undefined,
    }
  }
/>
`;
